### PR TITLE
Implement new drawer layout

### PR DESCRIPTION
**Pull request by theScrabi**
Implements #2204 

screenshot_1525437749
_____
**BUGS & IMPROVEMENTS**

- [X] (1) Quota does not show up in landscape mode. [FIXED]
- [ ] (2) Quota shows a not resolved template string if no quota was yet detected. [READY TO TEST]

### CODE REVIEW

File: src/com/owncloud/android/ui/activity/DrawerActivity.java [hunk 1]
```diff
@@ -236,6 +202,12 @@ public boolean onNavigationItemSelected(MenuItem menuItem) {
                                         ManageAccountsActivity.class);
                                 startActivityForResult(manageAccountsIntent, ACTION_MANAGE_ACCOUNTS);
                                 break;
+                            case R.id.drawer_menu_feedback:
+                                openFeedback();
```
Comment (davigonz) [inline src/com/owncloud/android/ui/activity/DrawerActivity.java]: These are brandeable options, so you should check `help_enabled` and `feedback_enabled` flags in setup.xml and hide or show the options in the side menu depending on that.

File: src/com/owncloud/android/ui/activity/DrawerActivity.java [hunk 2]
```diff
@@ -474,14 +468,21 @@ private void toggleAccountList() {
...
         if (mNavigationView != null) {
+            final int accountCount = AccountManager.get(this)
+                    .getAccountsByType(MainApp.getAccountType()).length;
+
             if (mIsAccountChooserActive) {
                 mAccountChooserToggle.setImageResource(R.drawable.ic_up);
                 mNavigationView.getMenu().setGroupVisible(R.id.drawer_menu_accounts, true);
                 mNavigationView.getMenu().setGroupVisible(R.id.drawer_menu_standard, false);
+                mNavigationView.getMenu().setGroupVisible(R.id.drawer_menu_settings_etc, false);
+                if(mDrawerCloud != null && accountCount > 4) mDrawerCloud.setVisibility(View.GONE);
```
Comment (davigonz) [inline src/com/owncloud/android/ui/activity/DrawerActivity.java]: What about creating a new constant with 4 value?

File: src/com/owncloud/android/ui/activity/Preferences.java [hunk 3]
```diff
@@ -426,7 +424,7 @@ public boolean onPreferenceClick(Preference preference) {
                     @Override
                     public boolean onPreferenceClick(Preference preference) {
                         String feedbackMail = (String) getText(R.string.mail_feedback);
-                        String feedback = (String) getText(R.string.prefs_feedback) +
+                        String feedback = (String) getText(R.string.drawer_feedback) +
```
Comment (davigonz) [inline src/com/owncloud/android/ui/activity/Preferences.java]: You could get rid of this preference

File: res/layout/drawer_header.xml [hunk 4]
```diff
@@ -21,96 +21,111 @@
...
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_gravity="bottom"
-        android:layout_alignParentBottom="true"
-        android:layout_alignParentRight="true"
-        android:layout_alignParentEnd="true"
-        android:contentDescription="@string/drawer_manage_accounts"
-        android:src="@drawable/ic_down"
-        />
+
```
[the remark this answers is not in the document]
Comment (theScrabi) [reply res/layout/drawer_header.xml]: I though the layout could fix the problem we where facing yesterday, however it seems we don't need it.

File: src/com/owncloud/android/media/MediaControlView.java [hunk 5]
```diff
@@ -123,7 +123,7 @@ private void initFloatingWindow() {
         mDecor = mWindow.getDecorView();
```
[the remark this answers is not in the document]
Comment (theScrabi) [reply src/com/owncloud/android/media/MediaControlView.java]: no changes should be applied here, that happened by accident.

File: src/com/owncloud/android/ui/activity/DrawerActivity.java [hunk 6]
```diff
@@ -250,11 +222,33 @@ public boolean onNavigationItemSelected(MenuItem menuItem) {
...
         if (mIsAccountChooserActive) {
             mNavigationView.getMenu().setGroupVisible(R.id.drawer_menu_accounts, true);
+
         } else {
             mNavigationView.getMenu().setGroupVisible(R.id.drawer_menu_accounts, false);
         }
     }
 
+    private void openHelp() {
+        final String helpWeb = (String) getText(R.string.url_help);
```
[the remark this answers is not in the document]
Comment (theScrabi) [reply src/com/owncloud/android/ui/activity/DrawerActivity.java]: Why not. It's good practice to declare variables that will not change their value as final.

File: src/com/owncloud/android/ui/activity/DrawerActivity.java [hunk 7]
```diff
@@ -250,11 +222,33 @@ public boolean onNavigationItemSelected(MenuItem menuItem) {
...
+        if (helpWeb != null && helpWeb.length() > 0) {
```
[the remark this answers is not in the document]
Comment (theScrabi) [reply src/com/owncloud/android/ui/activity/DrawerActivity.java]: no. copypasta error

File: src/com/owncloud/android/ui/activity/DrawerActivity.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (davigonz) [reply src/com/owncloud/android/ui/activity/DrawerActivity.java]: Agree but it is strange for me to use it whenever getting String resources. If you are going to include it, please use it in https://github.com/owncloud/android/pull/2215/files/ecc172270bb250e006bd3aa6f8292ebd66d009d6#diff-154af4ff2566e264dcca122072d20ae7R240 as well.

File: src/com/owncloud/android/ui/activity/Preferences.java (hunk 3 above)
Comment (theScrabi) [reply src/com/owncloud/android/ui/activity/Preferences.java]: As we spoke about it we came to the conclusion to keep this.

**Comment by davigonz** [comment]
@theScrabi changes approved, but there's still some conflicts in Preferences.java

**Comment by jesmrec** [comment]
In portrait the new drawer looks:

device-2018-05-18-121329

however, in landscape, the cloud pic you set in the bottom of the drawer does not appear

device-2018-05-18-121347

It is not a incorrect or bad behaviour, but you have to take in account that the user quota feature is also incoming, where the user should be able to check the quota at the bottom of the drawer, regardless the orientation of the device.

Any tweak here to be performed to make the quota visible? @theScrabi

**Comment by jesmrec** [comment]
Are the cloud pic and the background image in the top of the drawer brandable?

**Comment by theScrabi** [comment]
> however, in landscape, the cloud pic you set in the bottom of the drawer does not appear

Having the cloud in landscape would mead that it will be shown behind all the other images. However this is bad behavior, therefore I removed it there.

**Comment by jesmrec** [comment]
> Having the cloud in landscape would mead that it will be shown behind all the other images. However this is bad behavior, therefore I removed it there.

The cloud does not appear in landscape as an expected behaviour. But, the quota either. Is it something feasible to keep the qouta in the bottom of the drawer? we can face this issue in a separate feature.

**Comment by davigonz** [comment]
@jesmrec the bugs & improvements detected so far are already fixed

**Comment by jesmrec** [comment]
All stuff fixed. Approved

**Pull request merged**